scripts/rebase-ledger.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+exit_on_error() {
+    exit_code=$1
+    if [ $exit_code -ne 0 ]; then
+        mv ./data/ledger-temp.json ./data/ledger.json
+        exit $exit_code
+    fi
+}
+
+mv ./data/ledger.json ./data/ledger-temp.json
+exit_on_error $?
+
+git fetch
+exit_on_error $?
+
+git reset --hard origin/main
+exit_on_error $?
+
+node ./scripts/rebase-ledger.js
+exit_on_error $?
+
+rm ./data/ledger-temp.json
```

Comment:
this could/should be handled by the trap function, or used as a more naive condition. Like, if the temp file still exists, something went wrong, and the trap function should behave as it does now, rather than trying to handle specific exit codes.